SilMock/Google/Service/Directory/UsersResource.php

```diff
@@ -74,7 +73,76 @@ public function get($userKey)
 
         return $newUser;
     }
-
+    
+    /**
+     * @param $userKey
+     * @return Google_Service_Directory_Aliases|null
+     */
+    protected function getAliasesForUser($userKey)
+    {
+        // If the $userKey is not an email address, then it's an id.
+        $key = 'primaryEmail';
+        if (! filter_var($userKey, FILTER_VALIDATE_EMAIL)) {
+            $key = 'id';
+            $userKey = intval($userKey);
+        }
+        
+        $usersAliases = new UsersAliasesResource($this->_dbFile);
+        return $usersAliases->fetchAliasesByUser($key, $userKey);
+    }
+    
+    /**
+     * Get the database record of the user (if any) that has the given email
+     * address as an alias.
+     *
+     * NOTE: This does NOT do things like populate the returned user info with
+     * its list of aliases. That is left to the calling function (such as
+     * `UsersResource->get()`).
+     *
+     * @param $userKey
+     * @return null|Google_Service_Directory_User
+     */
+    protected function getDbUserByAlias($userKey)
+    {
+        if ( ! filter_var($userKey, FILTER_VALIDATE_EMAIL)) {
+            // This function only makes sense for actual email addresses.
+            return null;
+        }
+        
+        $allUsers = $this->getAllDbUsers();
+        
+        foreach ($allUsers as $aUser) {
+            if (! isset($aUser['data'])) {
+                continue;
+            }
+            
+            $userData = json_decode($aUser['data'], true);
+            if ($userData === null) {
+                continue;
+            }
+            
+            $primaryEmail = isset($userData['primaryEmail']) ? $userData['primaryEmail'] : null;
+            
+            $aliasesResource = $this->getAliasesForUser($primaryEmail);
+            if ($aliasesResource) {
+                foreach ($aliasesResource['aliases'] as $aliasResource) {
+                    $alias = $aliasResource['alias'];
+                    if (strcasecmp($alias, $userKey) === 0) {
```

Comment:
I just googled email case sensitivity and (re)learned something. Looks like this is technically correct.

Reply:
I at least did what I did intentionally. I _think_ it should be case insensitive, since I want to match an alias email address even if capitalization is different. To my knowledge, email addresses are generally considered case insensitive, though I bet not all email systems treat them that way.

Reply:
I at least did what I did intentionally. I _think_ it should be case insensitive, since I want to match an alias email address even if capitalization is different. To my knowledge, email addresses are generally considered case insensitive, though I bet not all email systems treat them that way.